src/Controller/Api/DatasetSubmissionController.php

```diff
@@ -421,17 +424,12 @@ public function validateUrlAction(int $id, Request $request, UrlValidation $urlV
      *
      * @return Response
      */
-    public function getGlobalIngestFolders(): Response
+    public function getGlobalIngestFolders(IngestUtil $ingestUtil): Response
     {
-        $user = $this->getUser();
-        /* TODO */
-        return $this->makeJsonResponse([
-            ".dotfolder",
-            "H1.x123.456.7890",
-            "H1.x123.456:7891",
-            "R2.x137.108.0003",
-            "Spaces in foldername",
-            "c1.x123.456:7890"
-        ]);
+        if (!($this->getUser() instanceof Account)) {
+            throw new AccessDeniedException('Must be logged in');
+        }
+        $jsonResponse = $this->makeJsonResponse($ingestUtil->getUsersIngestFoldersInIncomingDir($this->getUser()->getUserID()));
```

Comment:
```suggestion
        return $this->makeJsonResponse($ingestUtil->getUsersIngestFoldersInIncomingDir($this->getUser()->getUserID()));
```

Reply:
For some reason i'm not seeing the earlier fix in the file changes, this is the cleanest looking I think .